scripts/mixins/families/maat.lua

```diff
@@ -0,0 +1,113 @@
+require("scripts/globals/mixins")
+require("scripts/globals/status")
+
+
+g_mixins = g_mixins or {}
+
+g_mixins.maat = function(mob)
+ 
+    mob:addListener("SPAWN", "JOB_SPECIAL_SPAWN", function(mob)
+        if (mob:getMainJob() == dsp.job.NIN) then
+            mob:setLocalVar("specialThreshold", 40)
+        else
+            mob:setLocalVar("specialThreshold", math.random(50,60))
+        end
+    end)
+
+    mob:addListener("ROAM_TICK", "MAAT_RTICK", function(mob)
+        if mob:getLocalVar("engaged") == 0 then
+            for _, player in pairs(mob:getZone():getPlayers()) do
+                if player:checkDistance(mob) < 8 then
+                    local ID = require("scripts/zones/"..mob:getZoneName().."/IDs")
+                    mob:messageText(mob, ID.text.YOU_DECIDED_TO_SHOW_UP)
+                    mob:setLocalVar("engaged", player:getID())
+                end
+            end
+        end
+    end)
+
+    mob:addListener("ENGAGE", "MAAT_ENGAGE", function(mob, target)
+        if mob:getLocalVar("engaged") == 0 then
+            local ID = require("scripts/zones/"..mob:getZoneName().."/IDs")
+            mob:messageText(mob, ID.text.YOU_DECIDED_TO_SHOW_UP)
+            mob:setLocalVar("engaged", target:getID())
+        end
+    end)
+
+    mob:addListener("DISENGAGE", "MAAT_DISENGAGE", function(mob)
+        engagedID = mob:getLocalVar("engaged")
+        if engagedID ~= 0 then
+            player = GetPlayerByID(engagedID)
+            if player:getHP() == 0 then
+                local ID = require("scripts/zones/"..mob:getZoneName().."/IDs")
+                mob:showText(mob, ID.text.LOOKS_LIKE_YOU_WERENT_READY)
+            end
+        end
+    end)
+
+    mob:addListener("COMBAT_TICK", "MAAT_CTICK", function(mob)
+        local defaultAbility = {
+            [dsp.job.WAR] = dsp.jsa.MIGHTY_STRIKES_MAAT,
+            [dsp.job.MNK] = dsp.jsa.HUNDRED_FISTS_MAAT,
+            [dsp.job.WHM] = dsp.jsa.BENEDICTION_MAAT,
+            [dsp.job.BLM] = dsp.jsa.MANAFONT_MAAT,
+            [dsp.job.RDM] = dsp.jsa.CHAINSPELL_MAAT,
+            [dsp.job.THF] = dsp.jsa.PERFECT_DODGE_MAAT,
+            [dsp.job.PLD] = dsp.jsa.INVINCIBLE_MAAT,
+            [dsp.job.DRK] = dsp.jsa.BLOOD_WEAPON_MAAT,
+            [dsp.job.BST] = dsp.jsa.FAMILIAR_MAAT,
+            [dsp.job.BRD] = dsp.jsa.SOUL_VOICE_MAAT,
+            [dsp.job.RNG] = dsp.jsa.EES_MAAT,
+            [dsp.job.SAM] = dsp.jsa.MEIKYO_SHISUI_MAAT,
+            [dsp.job.NIN] = dsp.jsa.MIJIN_GAKURE_MAAT,
+            [dsp.job.DRG] = dsp.jsa.CALL_WYVERN_MAAT,
+            [dsp.job.SMN] = dsp.jsa.ASTRAL_FLOW_MAAT,
+        }
+
+        if mob:getHPP() < mob:getLocalVar("specialThreshold") then
+            local ID = require("scripts/zones/"..mob:getZoneName().."/IDs")
```

Comment:
You can get IDs from the zones global instead of doing string concats.
`local ID = zones[mob:getZoneID()]`

Reply:
Thanks for pointing that out. I was hoping there was a better way to do this!